### ISSUE TITLE
Replace incorrect usage of pclose() with fclose() in exec/exec.c

### ISSUE THREAD

**Issue by wdoekes** [OpenSIPS/opensips#1139]
TL;DR: The pclose()s in exec.c are wrong and should be replaced by fclose().

glibc popen() returns a custom FILE* that should be closed with pclose() according to the specs. It's like an fopencookie()-style FILE with custom close() handling, but not implemented with fopencookie(). In fact the glibc implementation uses the IO/FILE internals. [1]

(In reality, the glibc pclose() just calls fclose(), but that's glibc-specific. It *could* call something else. [2])

The popen3() as implemented in modules/exec/kill.c does not do special things to the FILE. It just returns fdopen()ed files. [3] So they should be closed with fclose(), not with pclose() [4].

[1] https://github.com/lattera/glibc/blob/a2f34833b1042d5d8eeb263b4cf4caaea138c4ad/libio/iopopen.c#L303
[2] https://github.com/lattera/glibc/blob/a2f34833b1042d5d8eeb263b4cf4caaea138c4ad/libio/pclose.c#L42
[3] https://github.com/OpenSIPS/opensips/blob/af584bc8f4667e8b6eb6b5aaf4923c8103fcf628/modules/exec/kill.c#L149
[4] https://github.com/OpenSIPS/opensips/blob/af584bc8f4667e8b6eb6b5aaf4923c8103fcf628/modules/exec/exec.c#L93

**Comment by liviuchircu** [comment]
It seems I started this trend 3 years ago (1ccdca46dd), and Ionut didn't spot this either during his own refactoring/additions to the module (3c2947e2cf5).

I just pushed a fix on `master` branch (96ba9ed03d2). Please let me know if it looks good, case in which we'll backport it and close this ticket.

**Comment by wdoekes** [comment]
LGTM

Hah, it was because of that 1ccdca4 that I was poking around here. When upgrading a box from 1.7 to 1.11, the exec_msg() stopped behaving like it used to: instead of closing stdin on the called job, it now waits for the job to stop.

I patched it here:
https://github.com/wdoekes/opensips/commit/af963817f386afbd8fd38060d98cce539c6943e7
But because it's so long ago, the new behaviour has probably settled. Therefore I did not file a PR.

I saw that the new exec() does not suffer from this problem, so the problematic code is deprecated anyway.
https://github.com/OpenSIPS/opensips/blob/96ba9ed03d2223f09b6fd25061d229c2c7ee5eb4/modules/exec/exec.c#L519

In any case. Thanks for the quick fix! :)

**Comment by liviuchircu** [comment]
Backported to `2.2` and `2.3`.

**Comment by wdoekes** [comment]
> We shouldn't change the behavior of a stable release, and those functions are deprecated anyway!

Agreed
